### PR TITLE
PM-10122: Autofocus on PIN or password field

### PR DESCRIPTION
**Pull request by shannon-livefront**
## 🎟️ Tracking

[PM-10122](https://bitwarden.atlassian.net/browse/PM-10122)

## 📔 Objective

Auto focus on the PIN or password entry fields when unlocking the vault or logging in.

## 📸 Screenshots

Logging in:
| Before | After |
| --- | --- |
| <video src="https://github.com/user-attachments/assets/84207a9a-350c-4fd6-b73d-fe30ab0b8d9d" /> | <video src="https://github.com/user-attachments/assets/c0fc252a-e33a-4aad-9950-5a3ca87563a0" /> |

Unlocking the vault:
| Before | After |
| --- | --- |
| <video src="https://github.com/user-attachments/assets/dab78f9a-4a64-4871-bd21-e1f502bc6e01" /> | <video src="https://github.com/user-attachments/assets/3b398fe2-fb9b-446d-a302-e45ee626cb71" /> |

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed
  issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

[PM-10122]: https://bitwarden.atlassian.net/browse/PM-10122?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: app/src/main/java/com/x8bit/bitwarden/ui/auth/feature/vaultunlock/VaultUnlockScreen.kt [hunk 1]
```diff
@@ -183,6 +183,7 @@ fun VaultUnlockScreen(
             ) {
                 if (!state.hideInput) {
                     BitwardenPasswordField(
+                        autoFocus = true,
```
Comment (david-livefront) [inline app/src/main/java/com/x8bit/bitwarden/ui/auth/feature/vaultunlock/VaultUnlockScreen.kt:186]: Do we want this to be conditional so it does not mess with the biometrics prompt that auto-appears when you have that set up?

```kotlin
autoFocus  = state.showBiometricLogin
```

Comment (dseverns-livefront) [reply app/src/main/java/com/x8bit/bitwarden/ui/auth/feature/vaultunlock/VaultUnlockScreen.kt:186]: I am not seeing the auto launch of the prompt happen that consistently (could be something about the SDK creating the cipher stuff that I don't understand) with or without these changes fwiw. I think I would favor this suggestion though as if I had biometrics enabled that would likely be my preferred way to login so the text field not being focused wouldnt be a big deal or something I expected.

**Review by SaintPatrck — COMMENTED** [review]
🤔 Unless I'm missing something, I believe this is addressing the same issue as #3601. @qwexter's changes respect biometrics and also prevent UnlockClick from processing if the input field is empty. Thoughts on using their changes and closing this PR?

**Comment by shannon-livefront** [comment]
> 🤔 Unless I'm missing something, I believe this is addressing the same issue as #3601. @qwexter's changes respect biometrics and also prevent UnlockClick from processing if the input field is empty. Thoughts on using their changes and closing this PR?

🫠 I'm glad you told me about that, thank you! I'm good with closing this PR

**Comment by SaintPatrck** [comment]
@shannon-livefront #3601 does not fix the issue on LoginScreen. Let's leave this PR open, but reduce the scope to focus on LoginScreen changes.

Also, I agree with @david-livefront and @dseverns-livefront that we should give biometrics precedence over password/pin field focus. This is how the MAUI app behaves, as well.

**Review by david-livefront — APPROVED** [review]

**Review by SaintPatrck — APPROVED** [review]

**Pull request merged**